### ISSUE TITLE
`multiline_arguments` incorrectly triggers when using Swift 6.1's trailing comma list feature

### ISSUE THREAD

**Issue by theblixguy** [realm/SwiftLint#6065]
### Bug Description

When `multiline_arguments` rule is enabled, it is triggered for the following example:

```swift
Test(
  foo: 1,
  bar: false,
)
```

Swift 6.1 adds the trailing comma list [feature](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0439-trailing-comma-lists.md) which means having a trailing comma in the above example is legal now.

I also noticed the following warning when linting:

```
warning: Skipping correcting file because it produced Swift parser errors: MyFile.swift
{
  "diagnostics" : [
    "expected value in function call"
  ]
}
```

### Environment

* SwiftLint version: 0.56.2
* Xcode version: Xcode 16.3
* Installation method used: Binary
* Configuration file:

```yml
opt_in_rules:
  multiline_arguments

multiline_arguments:
  first_argument_location: next_line
```

**Comment by SimplyDanny** [comment]
This should have been fixed from version 0.59.0 which includes the Swift 6.1 parser. Please check again with this newer version.
